### PR TITLE
[Relay][Op] Clip

### PR DESCRIPTION
**Pull request by joshpoll**
This PR registers the clip operator. I was wondering if I should put additional constraints on the attributes, though. `a_min` and `a_min` should be rank 0 tensors, which is not reflected in the current type relation. Additionally, should their data types match that of the input tensor `a`?

### CODE REVIEW

File: src/relay/op/tensor/unary.cc [hunk 1]
```diff
@@ -78,5 +78,37 @@ RELAY_REGISTER_OP("concat")
 .set_support_level(1)
 .add_type_rel("Concat", ConcatRel);
 
+// Clip
+struct ClipAttrs : public tvm::AttrsNode<ClipAttrs> {
+  // TODO: should I enforce these be 0 dim?
+  Constant a_min;
```
Comment (tqchen) [inline src/relay/op/tensor/unary.cc]: just use double as they are constant and double is enough to represent most numbers we care about

File: python/tvm/relay/op/tensor.py [hunk 2]
```diff
@@ -244,3 +244,35 @@ def concat(*args):
...
+
+
+def clip(a, a_min, a_max):
+    """Clip the elements in `a` between `min` and `max`.
+
+    Parameters
+    ----------
+    a : relay.Expr
+        The input tensor.
+    a_min : relay.Constant
```
Comment (tqchen) [inline python/tvm/relay/op/tensor.py]: let us just allow float

Comment (tqchen) [inline python/tvm/relay/op/tensor.py]: Although relay.Constant is more general in this case, let us just use float and make a_min and a_max attributes for now, this is the most common usecase of clip

Comment (joshpoll) [reply python/tvm/relay/op/tensor.py]: Do you mean that `a_min` and `a_max` should be python attributes? That deviates from numpy's function signature and also makes it impossible to enforce that `a_min` and `a_max` must be provided.

Comment (tqchen) [reply python/tvm/relay/op/tensor.py]: I mean make a_min and a_max as part of Attrs, see e.g. https://github.com/dmlc/tvm/blob/master/include/tvm/relay/attrs/transform.h#L17

**Comment by joshpoll** [comment]
@tqchen I addressed your feedback. My only lingering concern is that the data type of the tensor is changed by clip. I could write a custom type relation to express this.

**Comment by tqchen** [comment]
Why should data type of the tensor be changed by clip? I think we can enforce it to be the same as input for now,

**Comment by joshpoll** [comment]
Sorry. I was mistaken. I looked more closely at the clip implementation and I see that the bounds are cast to the input tensor's dtype.

**Pull request merged**